lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -397,6 +397,48 @@ class AwsCompileFunctions {
         delete functionResource.Properties.VpcConfig;
       }
 
+      const fileSystemConfigs = functionObject.fileSystemConfigs;
+
+      if (Array.isArray(fileSystemConfigs) && fileSystemConfigs.length === 1) {
+        const fileSystemConfig = fileSystemConfigs[0];
+
+        if (!functionResource.Properties.VpcConfig) {
+          const errorMessage = [
+            `Function "${functionName}": when using fileSystemConfigs, `,
+            'ensure that function has vpc configured ',
+            'on function or provider level',
+          ].join('');
+          throw new this.serverless.classes.Error(
+            errorMessage,
+            'LAMBDA_FILE_SYSTEM_CONFIG_MISSING_VPC'
+          );
+        }
+
+        const shouldUpdateIamRole =
+          typeof fileSystemConfig.arn === 'string' && fileSystemConfig.arn.startsWith('arn');
+        if (shouldUpdateIamRole) {
+          const iamRoleLambdaExecution = cfTemplate.Resources.IamRoleLambdaExecution;
+
+          const stmt = {
+            Effect: 'Allow',
+            Action: ['elasticfilesystem:ClientMount', 'elasticfilesystem:ClientWrite'],
+            Resource: [fileSystemConfig.arn],
+          };
+
```

Comment:
I think we should require arn string unconditionally, as with CloudFormation instructions we're unable to setup proper access rights.

We already do that in many other cases, so i think such restriction is not that bad (also in case of EFS it shouldn't be that problematic to rely only on ARN strings I think)

Reply:
So, do you suggest to totally skip support for `Fn::Import`, `Ref` and `Fn::GetAtt`?

Reply:
Yes, however if you see some significant issue in doing that, please elaborate

Reply:
That sounds reasonable, it's a bit clunkly to support it because, as you mentioned, it forces you to use custom Lambda Role with EFS policies, otherwise the function won't be able to interact with EFS at all. I removed support for it in the current iteration

Reply:
@pgrzesik here in 2024, what is the recommended way to use `Ref` or `Fn::GetAtt` here in `fileSystemConfig.arn`?  We are creating EFS as part of our `serverless.ts` (see below), but cannot reference the created `arn`.  Seems very limiting now in 2024:

```
const serverlessConfiguration = {
  custom: {
    esbuild: {
      external: ['canvas'],
      minify: true,
      packager: 'yarn',
      treeShaking: true,
    },
    localstack: {
      debug: false,
      edgePort: 4566,
      host: 'http://localhost',
      lambda: {
        mountCode: false,
      },
      stages: ['local'],
    },
    sentry: {
      dsn: serverlessEnv.SENTRY_DSN_LAMBDA,
    },
  },
  frameworkVersion: '3',
  functions: {
    clearDocumentSignatures: {
      handler: 'src/functions/clearDocumentSignatures.clearDocumentSignatures',
      name: 'clearDocumentSignatures-${opt:stage}',
      timeout: 600,
    },
    createDocumentVersions: {
      events: [
        { sqs: { arn: { 'Fn::GetAtt': ['CreateDocumentVersionQueue', 'Arn'] }, batchSize: 10 } },
      ],
      handler: 'src/functions/createDocumentVersions.createDocumentVersions',
      name: 'createDocumentVersions-${opt:stage}',
      timeout: 600,
    },
    extractFileJSON: {
      dependsOn: ['FileSystem'],
      fileSystemConfig: {
        arn: 'FileSystem.arn',
        localMountPath: '/mnt/tmp',
      },
      handler: 'src/functions/extractFileJSON.extractFileJSON',
      name: 'extractFileJSON-${opt:stage}',
      timeout: 600,
    },
    flushDocumentVersions: {
      handler: 'src/functions/flushDocumentVersions.flushDocumentVersions',
      name: 'flushDocumentVersions-${opt:stage}',
      timeout: 600,
    },
    scheduleCreateDocumentHistoryVersions: {
      environment: {
        // Seems like this injects CREATE_DOCUMENT_VERSION_QUEUE_URL onto process.env...
        CREATE_DOCUMENT_VERSION_QUEUE_URL: { Ref: 'CreateDocumentVersionQueue' },
      },
      events: [{ schedule: { rate: 'rate(1 minute)' } }],
      handler:
        'src/functions/scheduleCreateDocumentHistoryVersions.scheduleCreateDocumentHistoryVersions',
      name: 'scheduleCreateDocumentHistoryVersions-${opt:stage}',
      timeout: 600,
    },
    setDocumentState: {
      handler: 'src/functions/setDocumentState.setDocumentState',
      name: 'setDocumentState-${opt:stage}',
      timeout: 600,
    },
  },
  plugins: ['serverless-esbuild', 'serverless-localstack', 'serverless-sentry'],
  provider: {
    environment: serverlessEnv,
    name: 'aws',
    runtime: 'nodejs16.x',
    vpc: {
      securityGroupIds: isNoVPCNeeded ? [] : securityGroupIds,
      subnetIds: isNoVPCNeeded ? [] : subnetIds,
    },
  },
  resources: {
    Resources: {
      CreateDocumentVersionQueue: {
        Properties: { QueueName: 'create-document-version-${opt:stage}', VisibilityTimeout: 3600 },
        Type: 'AWS::SQS::Queue',
      },
      FileSystem: { Properties: {}, Type: 'AWS::EFS::FileSystem' },
      FileSystemAccessPoints: {
        DependsOn: 'FileSystemMountTarget',
        Properties: {
          FileSystemId: { Ref: 'FileSystem' },
          Path: '/',
          PosixUser: { Gid: '1000', Permissions: 777, Uid: '1000' },
          RootDirectory: { CreationInfo: { OwnerGid: '1000', OwnerUid: '1000', Permissions: 755 } },
        },
        Type: 'AWS::EFS::AccessPoint',
      },
      FileSystemMountTarget: {
        Properties: {
          FileSystemId: { Ref: 'FileSystem' },
          SecurityGroups: securityGroupIds,
          SubnetId: serverlessEnv.AWS_SUBNET_PUBLIC_ID,
        },
        Type: 'AWS::EFS::MountTarget',
      },
    },
  },
  service: 'collab-worker',
} satisfies Serverless;
```